administrator/components/com_joomgallery/helpers/file.php

```diff
@@ -367,301 +383,409 @@ public static function resizeImage(&$debugoutput, $src_file, $dest_file, $usefor
       return true;
     }
 
-    // For free resizing and cropping the center
-    $offsetx = null;
-    $offsety = null;
-    if($max_width)
-    {
-      $debugoutput .= JText::_('COM_JOOMGALLERY_UPLOAD_RESIZE_TO_MAX').'<br />';
-
-      if($new_width <= 0)
+    // determine resizing width and height
+      $offsetx = null;  
+      $offsety = null;
+      $debugoutput .= JText::_('COM_JOOMGALLERY_UPLOAD_CREATE_THUMBNAIL_FROM').' '.$imginfo['type'];
+      if ($anim && $imginfo['animation'])
       {
-        $debugoutput .= JText::_('COM_JOOMGALLERY_UPLOAD_ERROR_NO_VALID_WIDTH_OR_HEIGHT').'<br />';
-
-        return false;
+        $debugoutput .= ' '.JText::_('COM_JOOMGALLERY_UPLOAD_CREATE_WITH_ANIM');
       }
+      elseif ($imginfo['transparency'])
+      {
+        $debugoutput .= ' '.JText::_('COM_JOOMGALLERY_UPLOAD_CREATE_WITH_TRANS');
+      }
+      elseif ($anim && $imginfo['animation'] && $imginfo['transparency'])
+      {
+        $debugoutput .= ' '.JText::_('COM_JOOMGALLERY_UPLOAD_CREATE_WITH_TRANS_ANIM');
+      }
+      $debugoutput .= ', '.$imginfo['width'].' x '.$imginfo['height'].'...<br />';
 
-      $ratio = max($srcHeight,$srcWidth) / $new_width ;
```

Comment:
calculation of ratio by the max_width argument is replaced by case0 and case1 of the switch below...